Sarthak iP

Halpfrog opened the pull request:
# Duke
>This is just a quote by me -Sarthak
Duke is going to be best assistant for master chief
-Add a task
-~~Remove~~ Adjust completed events
-Clear out all _banished_
All you need to do _is_

1. Get cortana [working](https://pages.github.com/)
2. Jump off the pelican
3. Create activities

Best part is that it already hates the **Banished**

- [x] Destroy the FOB
- [ ] Rescue the pilot
- [ ] Stop the rebuilding :tada:

If you use Java why dont you just edit <code>main</code> file
> Here is the file
>
> <code>public class Main { </code>
        <code> public static void main(String[] args) { </code>
        <code> Application.launch(MainApp.class, args); </code>

Commit:
Add Gradle support

Halpfrog committed:
Create chatbot.java

added level 1

Halpfrog committed:
Level 2

Halpfrog committed:
level-3

Halpfrog committed:
Level - 4

Halpfrog committed:
added auto test functionality

Halpfrog committed:
Added exception handling

Halpfrog committed:
added delete functionality

Halpfrog committed:
Removed class files

Halpfrog committed:
Updated Duke to contain features of chatbot

dandaandaaaaaan commented on src/main/java/Deadline.java:
```diff
@@ -0,0 +1,13 @@
+public class Deadline extends Task{
```
```suggestion
public class Deadline extends Task {
```
For coding standard

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+            try { 
+                Scanner sc = new Scanner(System.in);
+                Task n = new Task(sc.nextLine());
+                if(n.description.equals("bye")) {
```
```suggestion
                if (n.description.equals("bye")) {
```
For coding standard

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                } else {
+                    throw new DukeException("I dont know what you want but this aint it");
+                }
+            } catch (DukeException error){
```
```suggestion
            } catch (DukeException error) {
```
For coding standard

dandaandaaaaaan commented on src/main/java/Event.java:
```diff
@@ -0,0 +1,12 @@
+public class Event extends Task{
```
```suggestion
public class Event extends Task {
```
For coding standard

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                    break;
+                } else if (n.description.equals("list")) {
+                    System.out.println("Here are the things you've tried to do\n");
+                    for(int i = 1; i <= ls.size(); i++) {
```
```suggestion
                    for (int i = 1; i <= ls.size(); i++) {
```
For coding standard

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                    Task edited = ls.remove(x -1);
+                    edited.done();
+                    ls.add(x-1, edited);
+                    for(int i = 1; i <= ls.size(); i++) {
```
```suggestion
                    for(int i = 1; i <= ls.size(); i++) {
```
For coding standard

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                    Task edited = ls.remove(x -1);
+                    edited.undone();
+                    ls.add(x-1, edited);
+                    for(int i = 1; i <= ls.size(); i++) {
```
```suggestion
                    for(int i = 1; i <= ls.size(); i++) {
```
For coding standard

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                    }
+                } else if(n.description.startsWith("todo")) {
+                    todo a = new todo(n.description.substring(5));
+                    if(a.description.equals("")) {
```
```suggestion
                    if (a.description.equals("")) {
```
For coding standard

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                        System.out.println(i + " " + ls.get(i -1));
+                    }
+                } else if(n.description.startsWith("todo")) {
+                    todo a = new todo(n.description.substring(5));
```
```suggestion
                    todo newTask = new todo(n.description.substring(5));
```
For coding standard, to give more meaningful variable names

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                    System.out.println("New list size is " + ls.size());
+                } else if(n.description.startsWith("delete")) {
+                    String number = n.description.substring(7);
+                    int x = Integer.parseInt(number);
```
```suggestion
                    int taskNumber = Integer.parseInt(number);
```
For coding standard, to give more meaningful variable names

dandaandaaaaaan commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+        while(true) {
+            try { 
+                Scanner sc = new Scanner(System.in);
+                Task n = new Task(sc.nextLine());
```
```suggestion
                Task task = new Task(sc.nextLine());
```
For coding standard, to avoid single letter variables unless as scratch variables

dandaandaaaaaan commented on src/main/java/chatbot.java:
```diff
@@ -0,0 +1,81 @@
+import java.util.Scanner;
```
Will not highlight for this file as it seems to be largely duplicated with `Duke.java`

dandaandaaaaaan reviewed (commented):
I found your code easy to follow, but many things such as the UI and commands can be further extracted through OOP. I also tried to spot coding standard violations and documented them.

Halpfrog committed:
Added write to txt options

TypeDefinition commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                    "What can I do for you? \n";
+        System.out.println(intro);
+        List<Task> ls = new ArrayList<>();
+        while(true) {
```
As per the coding standard, consider adding a space between `while` and `(true)`, so that it becomes `while (true)`.

TypeDefinition commented on src/main/java/Duke.java:
```diff
@@ -6,5 +9,78 @@ public static void main(String[] args) {
+                } else if(n.description.startsWith("todo")) {
+                    todo a = new todo(n.description.substring(5));
+                    if(a.description.equals("")) {
+                        throw new DukeException("Nothings here");
+                    } else {
+                        ls.add(a);
+                        System.out.println("Added the following todo");
+                        System.out.println("added: " + a);
+                        System.out.println("New list size is " + ls.size());
+                    }
+                } else if(n.description.startsWith("deadline")) {
+                    String[] parts = n.description.substring(9).split(" /by ");
+                    Deadline d = new Deadline(parts[0], parts[1]);
+                    ls.add(d);
+                    System.out.println("Added the following todo");
+                    System.out.println("added: " + d);
+                    System.out.println("New list size is " + ls.size());
+                } else if(n.description.startsWith("event")) {
+                    String[] parts = n.description.substring(6).split(" /at ");
+                    Event e = new Event(parts[0], parts[1]);
+                    ls.add(e);
+                    System.out.println("added: " + e);
+                    System.out.println("New list size is " + ls.size());
+                } else if(n.description.startsWith("delete")) {
+                    String number = n.description.substring(7);
+                    int x = Integer.parseInt(number);
+                    Task item = ls.remove(x-1);
+                    System.out.println("I've removed the item from the list, stop being so lazy");
+                    System.out.println(item);
+                    System.out.println("you now have " + ls.size() + " things to do");
+                } else {
+                    throw new DukeException("I dont know what you want but this aint it");
+                }
+            } catch (DukeException error){
+                System.out.println("OOPS! " + error.getMessage());
+            } catch (IndexOutOfBoundsException a) {
+                System.out.println("Nothing inside");
+            }
+
+        } 
```
This `while` loop tends to become very long and bloated as the number of commands increase. Perhaps consider converting each `if` condition into a command class, and store them into a `HashMap` where you can easily search each command via its trigger word.

TypeDefinition commented on src/main/java/chatbot.java:
```diff
@@ -0,0 +1,81 @@
+                    throw new DukeException("Nothings here");
+                } else {
+                    ls.add(a);
+                    System.out.println("Added the following todo");
+                    System.out.println("added: " + a);
+                    System.out.println("New list size is " + ls.size());
+                }
+            } else if(n.description.startsWith("deadline")) {
+                String[] parts = n.description.substring(9).split(" /by ");
+                Deadline d = new Deadline(parts[0], parts[1]);
+                ls.add(d);
+                System.out.println("Added the following todo");
+                System.out.println("added: " + d);
+                System.out.println("New list size is " + ls.size());
+            } else if(n.description.startsWith("event")) {
+                String[] parts = n.description.substring(6).split(" /at ");
+                Event e = new Event(parts[0], parts[1]);
+                ls.add(e);
+                System.out.println("added: " + e);
+                System.out.println("New list size is " + ls.size());
+            } else if(n.description.startsWith("delete")) {
+                String number = n.description.substring(7);
+                int x = Integer.parseInt(number);
+                Task item = ls.remove(x-1);
+                System.out.println("I've removed the item from the list, stop being so lazy");
+                System.out.println(item);
+                System.out.println("you now have " + ls.size() + " things to do");
+            } else {
+                throw new DukeException("I dont know what you want but this aint it");
+            }
+        } catch (DukeException error){
+            System.out.println("OOPS! " + error.getMessage());
+        } catch (IndexOutOfBoundsException a) {
+            System.out.println("Nothing inside");
+        }
+
+    } 
+    System.out.println("Bye. Hope to see you again");
+    
+    }
```
This file looks very similar to `Duke.java`. Is it possible to combine them?

TypeDefinition reviewed (commented):
Good job! Overall the code was very easy to understand. :+1:  However, consider combining `Duke.java` and `chatbot.java` as they seem to be doing very similar things. Also, as per the coding standard, all classes and filenames should be in PascalCase.

Halpfrog committed:
added date and time

Halpfrog committed:
Merge branch 'branch-Level-8'

Halpfrog committed:
refactored

Halpfrog committed:
packaging added

Halpfrog committed:
fixed date time errors

Halpfrog committed:
Added Junit tests

Halpfrog committed:
Added javadocs

Halpfrog committed:
Fixed according to coding standard

Halpfrog committed:
Added find functionality

Halpfrog committed:
Merge remote-tracking branch 'origin/add-gradle-support'

Halpfrog committed:
Added a Gui

Halpfrog committed:
Fix find Method to only return a result for a word not a substring

Previously find would return a result
even if only a substring was found not only the word.

Added the use of Assert to ensure proper addition of tasks

Ensures that addition of tasks has been done correctly

Refactoring use method in parser to return a String

This is to make sure that the output of the message can be seen by the user while using the GUI.

Commit:
Fix find Method to only return a result for a word not a substring

Previously find would return a result
even if only a substring was found not only the word.

Added the use of Assert to ensure proper addition of tasks

Ensures that addition of tasks has been done correctly

Refactoring use method in parser to return a String

This is to make sure that the output of the message can be seen by the user while using the GUI.

Halpfrog committed:
Lose Unnecessary code in Duke.

Halpfrog committed:
Remove unnecessary code from Duke.java

Remove unnecessary code from Task.java

Halpfrog committed:
Merge tag 'A-CodeQuality'

Halpfrog committed:
Merge pull request #2 from Halpfrog/A-Assertions

A assertions

Halpfrog committed:
Add flexibility to Find function to allow for partial matches

Halpfrog committed:
Change Gui appearence

Halpfrog committed:
Edit Gui

Halpfrog committed:
Add background image and changed duke image

Halpfrog committed:
Add background image and changed duke image

Halpfrog committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/resources/view/MainWindow.fxml

Halpfrog committed:
Add Ui.png for screenshot

Halpfrog committed:
Add Ui.png for screenshot

Halpfrog committed:
Change Date time

Halpfrog committed:
Merge branch 'branch-A-JavaDoc'

Halpfrog committed:
Fix unnecessary code

Halpfrog committed:
Merge branch 'branch-A-JavaDoc'

Halpfrog committed:
Fix unnecessary code

Halpfrog committed:
Fix unnecessary code

Halpfrog committed:
Merge branch 'branch-A-CodingStandard'

Halpfrog committed:
Fix unnecessary code

Halpfrog committed:
Merge branch 'branch-A-JavaDoc'

Halpfrog committed:
Revert "Fix unnecessary code"

This reverts commit 1da9f1d3e2b1ee62b24f4f79401623fb988528ab.

Halpfrog committed:
Revert "Fix unnecessary code"

This reverts commit e0d37e43c254dd7e3dc8ee1997fc89b2b76e5249.

Halpfrog committed:
Fix naming of methods

Halpfrog committed:
Merge branch 'branch-A-CodingStandard'

# Conflicts:
#	src/main/java/Duke/Processing/Parser.java

Halpfrog committed:
Add User Guide

Halpfrog committed:
Make time easier to read

Fix issue with mark and unmark not having visible results

Add working scroll bar

Halpfrog committed:
Fix Junit tests to account for different date format

Halpfrog committed:
Fix Junit tests to account for different date format

Halpfrog committed:
Merge branch 'testing_date'

Halpfrog committed:
Edit Datetime error format

Change visible reactions to mark and unmark

Delete unnecessary images

Halpfrog committed:
Change Gui for DialogBox

Made the images circular
Allow the dialog box to expand if more items are in the tasklist
